lib/url-pattern/url-match-pattern.js

```diff
@@ -64,19 +77,26 @@ _.assign(UrlMatchPattern.prototype, /** @lends UrlMatchPattern.prototype */ {
      */
     createMatchPattern: function () {
         var matchPattern = this.pattern,
+            match;
+
+        // Check the match pattern of sanity and split it into protocol, host and path
 
-            // Check the match pattern of sanity and split it into protocol, host and path
+        if (this._allowProtocols) {
```

Comment:
Coz here I needed an `else` block also. All the above cases are just `if-then`

Using that pattern here will make it very un-readable
```js
(this._allowProtocols && (match = matchPattern.match(regexes.noProtocolPatternSplit);) )
 || (match = matchPattern.match(regexes.patternSplit);)
```